### DISCUSSION TITLE
[WinError 2] The system cannot find the file specified

### DISCUSSION THREAD

**Discussion by jccastro94** [openai/whisper#109] [Q&A] [answered]
I was just testing it out and couldn't get any audio, no matter how long, to transcribe.
The Python error output is too long, so here is the command line output:

```
Traceback (most recent call last):
  File "D:\Whisper\.venv\Scripts\whisper-script.py", line 33, in <module>
    sys.exit(load_entry_point('whisper==1.0', 'console_scripts', 'whisper')())
  File "D:\Whisper\.venv\lib\site-packages\whisper\transcribe.py", line 275, in cli
    result = transcribe(model, audio_path, temperature=temperature, **args)
  File "D:\Whisper\.venv\lib\site-packages\whisper\transcribe.py", line 76, in transcribe
    mel = log_mel_spectrogram(audio)
  File "D:\Whisper\.venv\lib\site-packages\whisper\audio.py", line 111, in log_mel_spectrogram
    audio = load_audio(audio)
  File "D:\Whisper\.venv\lib\site-packages\whisper\audio.py", line 42, in load_audio
    ffmpeg.input(file, threads=0)
  File "D:\Whisper\.venv\lib\site-packages\ffmpeg\_run.py", line 313, in run
    process = run_async(
  File "D:\Whisper\.venv\lib\site-packages\ffmpeg\_run.py", line 284, in run_async
    return subprocess.Popen(
  File "C:\Users\JC\AppData\Local\Programs\Python\Python39\lib\subprocess.py", line 951, in __init__
    self._execute_child(args, executable, preexec_fn, close_fds,
  File "C:\Users\JC\AppData\Local\Programs\Python\Python39\lib\subprocess.py", line 1420, in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2] The system cannot find the file specified

```

So the closest error I assume it is, is a subprocess.Popen() error, but I'm not sure how to solve it. It's something on the whisper library side, cause on my side, this is my simple code:

```
import whisper

model = whisper.load_model("base.en")
audio = "audios/Project_Thomas.mp3"
fileexists = os.path.isfile(audio)
print(fileexists)
result = model.transcribe(audio, fp16=False, language="en")
```

any thoughts?

**Comment by jccastro94** [comment, marked as answer]
I was using the wrong environment, one without ffmpeg. Seems it couldn't find it. 
Glad it was simple though it did give me a headache. XD

Answer: Ensure you have ffmpeg
$ choco install ffmpeg

**Reply by tonirojas** [reply]
I tried this, in venv enviroments and local ones, with different python versions (3.8, 3.9, 3.10), with ffmpeg installed but allways having the same error: 
...in _execute_child
    hp, ht, pid, tid = _winapi.CreateProcess(executable, args,
FileNotFoundError: [WinError 2]

The thing is, I am trying to execute whisper openai but when I send model.device the result is always CPU. I think thats the problem but not sure 100% (Cuda 11 installed on windows 11)

**Reply by ccmilne** [reply]
Also struggling with these issues. Still receiving "FileNotFoundError: [WinError 2] The system cannot find the file specified" when trying to transcribe. 

I'm working in a virtual env with python 3.8 and installed ffmpeg via ubuntu. Also tried pip installing ffmpeg within the venv. Any ideas?

**Reply by emanrow** [reply]
@tonirojas @ccmilne I ran into this as well, trying to use whisper in a Jupyter notebook in VS Code. I had ffmpeg installed in Windows, and in my python environment, but whisper requires it in the Windows VM that VS Code uses to run the notebook, which I usually don't interact with.

There may be some slick way to add ffmpeg to the path of that environment after installing it in the terminal VS Code provides, but I couldn't find it. Instead, I just copied ffmpeg.exe from the bin of my Windows install to the folder where my Python notebook lives.

**Reply by iamAbdullahKhalid** [reply]
@emanrow am glad it also worked for me. I had also installed the ffmpeg but the issue was still there till i copied ffmpeg.exe and ffmpeg.shim to the directory where .ipynb was present

**Reply by Juan-Fruto** [reply]
@emanrow and @iamAbdullahKhalid how/where do you get the ffmpeg executable?

**Reply by iamAbdullahKhalid** [reply]
> @emanrow and @iamAbdullahKhalid how/where do you get the ffmpeg executable?

you can get it by searching on windows search bar on the bottom left corner. Just simply search "ffmpeg.exe" the file will pop up ,by right clicking choose the option "open file location". From there you can copy both the files.

**Reply by danialn1** [reply]
There is no ffmpeg.exe in my system how to download

**Reply by Juan-Fruto** [reply]
@danialn1 i downloaded it from here https://www.videohelp.com/software/ffmpeg

Just copy the ffmpeg.exe to the directory of yor project.

**Reply by mechXsteam** [reply]
Thank you so much! If it weren't for your comment, I wouldn't have been able to resolve my issue.

**Reply by zynu007** [reply]
thank you. this solved the error for me.  I was using the tiny model for an assignment and was stuck by the error for hours. :)

**Reply by shubh637** [reply]
can anyone help how i can solve this problem i am using mp4 file 

 model = whisper.load_model("base")  # Use the Whisper model
        result = model.transcribe(video_path)
        return result["text"]                    

this is my code

**Comment by architectdrone** [comment]
this helped me, thanks :)

**Comment by caytienkuagai99** [comment]
thank you, I have installed ffmpeg and can use whisper

**Comment by henryleo180** [comment]
Thank you! And here I find for my case, pip install ffmpeg/ffmpeg-python doesn't work but choco work!

**Comment by Macdaddy4sure** [comment]
You can also try running the program as an administrator / root.

**Comment by KD7JHD** [comment]
Why doesn't pip install ffmpeg-python work? I already know I'm in the correct venv.
